File: solved/155A.rb
def count(a)
  tot_amazing = 0
  # The first content is the max and the min number of points
  min = a.shift
  max = min
  a.each do |p|
    if p < min
      min = p
      tot_amazing += 1
    elsif p > max
      max = p
      tot_amazing += 1
    end
  end
  tot_amazing
end

n = gets.to_i
a = gets.split(/\s/).map(&:to_i)

puts count(a)

File: solved/155A.cc
#include <iostream>

using namespace std;

int main() {
  int n, min, max;
  cin >> n >> min;
  max = min;
  int ans = 0;
  for(int i=0; i<n-1; ++i) {
    int e;
    cin >> e;
    if(e<min) {
      ans++;
      min = e;
    }
    if(e>max) {
      ans++;
      max = e;
    }
  }
  cout << ans;
  return 0;
}
